src/components/MapContainer.js

```diff
@@ -80,12 +80,12 @@ class MapContainer extends React.Component {
       .then(coordinates => getTraceAttributes(host, coordinates))
       // This `catch` statement is placed here to handle errors from Fetch API.
       .catch(error => {
+        this.props.stopLoading()
```

Comment:
By moving it here, it now only stops loading if an error occurs -- it still displays the loader if the request is successful.

One thought might be to consider that certain actions (perhaps this error is one of them) clears all current loading actions, even if there are multiple in the counter.